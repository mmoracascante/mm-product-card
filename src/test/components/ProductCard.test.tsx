import React from "react";
import renderer from 'react-test-renderer'
import { ProductCard } from '../../components'
import { product2 } from '../data/products'


const { act } = renderer



describe('Testeando componente <ProductCard />', () => {

    test('Debe mostrar el compoenent correctamente', () => {
        const wrapper = renderer.create(
            <ProductCard product={product2}>

                {
                    () => (
                        <h1>Product Card</h1>
                    )
                }
            </ProductCard>



        )
        expect(wrapper.toJSON()).toMatchSnapshot()
    })

    test('Debe incrementar el contador', () => {
        const wrapper = renderer.create(
            <ProductCard product={product2}>

                {
                    ({ count, increaseBy }) => (
                        <>
                            <h1>Product Card</h1>
                            <span>{count}</span>
                            <button onClick={() => increaseBy(+1)}></button>
                        </>
                    )
                }
            </ProductCard>
        )

        let tree = wrapper.toJSON()
        expect(tree).toMatchSnapshot();
        act(() => {
            (tree as any).children[2].props.onClick()

        })

        tree = wrapper.toJSON()
        expect((tree as any).children[1].children[0]).toBe('1');

        // console.log(tree)
    })
})