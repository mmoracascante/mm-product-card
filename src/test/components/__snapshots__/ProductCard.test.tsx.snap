// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testeando componente <ProductCard /> Debe incrementar el contador 1`] = `
<div
  className="productCard undefined"
>
  <h1>
    Product Card
  </h1>
  <span>
    0
  </span>
  <button
    onClick={[Function]}
  />
</div>
`;

exports[`Testeando componente <ProductCard /> Debe mostrar el compoenent correctamente 1`] = `
<div
  className="productCard undefined"
>
  <h1>
    Product Card
  </h1>
</div>
`;
